### DISCUSSION TITLE
res.revalidate and res.setPreviewData with Next 13.2 app Route Handlers

### DISCUSSION THREAD

**Discussion by wimbarelds** [vercel/next.js#46475] [Help]
With Route Handlers inside the app directory introduced in Next 13.2, I am trying to migrate existing `pages/api` routes, but having trouble finding replacements for `res.revalidate`, `res.setPreviewData` and `res.clearPreviewData` that are available on `NextApiResponse`, but not on `NextResponse` that is used in the app-directory Route Handlers.

Are these features not yet supported? Not yet documented? Am I overlooking something?

**Comment by icyJoseph** [comment]
I don't think those exist just yet... 

- https://beta.nextjs.org/docs/api-reference/response

Since that NextResponse object comes from `next/server`, meaning they are closer to the API's available in middleware, might even be the same, than they are to the `next` exported `NextApiResponse`, and that's because they build on top of Response and Request from the Web API, whereas the objects in pages/api are built on top of Node built-ins (ServerResponse and IncomingRequest)

```ts
import type { NextApiRequest, NextApiResponse } from 'next'
```

vs

```ts
import { NextRequest, NextResponse } from "next/server";
```
